### PR TITLE
19 promisify apis

### LINKED ISSUE #19: Promisify APIs
__What should we do?__

We should upgrade `moddle-xml@9` to use the latest Promise support. Therefore the API of this project has to be promisified as well.

__Why should we do it?__

_Original text_

There seems to be usage of setTimeout() in 8.x of moddle-xml, which causes issues in non-browser and non-node execution envs.  The usage seems to have been removed from the 9x version of of moddle-xml.  

Is there a timeline for dmn-moddle to be moved to 9.x of moddle-xml?

----

Related to https://github.com/bpmn-io/bpmn-moddle/pull/73

### PR DESCRIPTION
**Pull request by MaxTru**
closes #19

### CODE REVIEW

File: test/spec/xml/roundtrip.js [hunk 1]
```diff
@@ -62,45 +62,38 @@ describe('dmn-moddle - roundtrip', function() {
...
-        const { warnings } = context;
+    const file = fs.readFileSync(fileName, 'utf8');
 
-        try {
-          if (process.env.VERBOSE && warnings.length > 0) {
-            console.log('import warnings', warnings);
-          }
+    const {
+      rootElement: definitions,
+      warnings } = await moddle.fromXML(file, 'dmn:Definitions');
```
Comment (nikku) [inline test/spec/xml/roundtrip.js:87]:
```suggestion
      warnings
    } = await moddle.fromXML(file, 'dmn:Definitions');
```

File: test/integration/tck.js [hunk 2]
```diff
@@ -73,36 +74,25 @@ describe('dmn-moddle - TCK roundtrip', function() {
...
 
         // when
-        moddle.fromXML(fileContents, 'dmn:Definitions', function(err, result, context) {
-
-          if (err) {
-            return done(err);
-          }
+        const {
+          rootElement: definitions,
+          warnings } = await moddle.fromXML(fileContents, 'dmn:Definitions');
```
Comment (nikku) [inline test/integration/tck.js:79]:
```suggestion
          warnings
        } = await moddle.fromXML(fileContents, 'dmn:Definitions');
```

**Review by nikku — CHANGES_REQUESTED** [review]
Two small style adjustments. Otherwise looking great :clap:

File: test/integration/tck.js (hunk 2 above)
Comment (MaxTru) [reply test/integration/tck.js]: :+1:

**Pull request merged**